src/bp/ui-studio/src/web/views/OneFlow/diagram/RouterForm/index.tsx

```diff
@@ -0,0 +1,173 @@
+import { Tab, Tabs } from '@blueprintjs/core'
+import axios from 'axios'
+import { FlowNode, FlowVariable, FlowVariableOperator } from 'botpress/sdk'
+import { Contents, Dropdown, lang, MoreOptions, MoreOptionsItems, Option, RightSidebar } from 'botpress/shared'
+import { Variables } from 'common/typings'
+import React, { FC, Fragment, useEffect, useState } from 'react'
+import { condition } from '~/views/FlowBuilder/common/style.scss'
+
+import style from './style.scss'
+import { Operation } from './Operation'
+import { OperationParser } from './OperationParser'
+import { OperationSerializer } from './OperationSerializer'
+
+interface Props {
+  deleteNode: () => void
+  close: () => void
+  node: FlowNode
+  diagramEngine: any
+  variables: Variables
+  editingCondition: number
+  contentLang: string
+  onUpdateVariables: (variable: FlowVariable) => void
+}
+
+const RouterForm: FC<Props> = ({
+  close,
+  node,
+  diagramEngine,
+  deleteNode,
+  variables,
+  editingCondition,
+  contentLang,
+  onUpdateVariables
+}) => {
+  const transition = node?.next?.[editingCondition]
+  const [showOptions, setShowOptions] = useState(false)
+
+  const parseInitialOperation = (): Operation => {
+    const parser = new OperationParser()
+    try {
+      return parser.parse(transition?.condition)
+    } catch {
+      return { args: {} } as Operation
+    }
+  }
+
+  const [operation, setOperation] = useState<Operation>(parseInitialOperation())
+  const variableType = variables?.currentFlow?.find(x => x.params.name === operation.variable)?.type
+  const variableConfig = variables?.primitive?.find(x => x.id === variableType)?.config
+  const operatorOptions = variableConfig?.operators?.map<Option>(x => ({ label: lang.tr(x.label), value: x.func }))
+  const operator = variableConfig?.operators?.find(x => x.func === operation.operator)
```

Comment:
variables.primitive will always be available, you can check null check for variables.currentFlow?. however.